uaclient/config.py

```diff
@@ -285,12 +288,16 @@ def _attached_status(self) -> "Dict[str, Any]":
                 contractInfo["effectiveTo"], "%Y-%m-%dT%H:%M:%SZ"
             )
 
-        resources = get_available_resources(self)
-        inapplicable_resources = [
-            resource["name"]
+        resources = self.machine_token.get("availableResources", [])
```

Comment:
If the contract server isn't yet producing `availableResources`, how can we be sure that the logic handling it is correct? I'm concerned that there may be some difference between our expectations and reality, which we may only discover once this branch has landed (and, potentially, released which would obviously be much worse).

If we don't have reason to be very confident in the logic here (please tell me if we do!), then I think we should wait until the contract server is actually producing this content to write code that relies on it.

Reply:
@OddBloke, right now the contract server **is** surfacing no availableResource key from the initial attached machine-token from the route **/v1/context/machines/token**.  But, that availableResources key is absent from the refreshed machine-token.json which we get from **/v1/resources/{resource}/context/machines/{machine}**.

The content from the initial attach 
```
   "availableResources" : [
      {
         "name" : "cc-eal",
         "available" : false
      },
      {
         "name" : "esm-infra",
         "available" : true
      },
      {
         "available" : false,
         "name" : "fips"
      },
      {
         "name" : "fips-updates",
         "available" : false
      },
      {
         "name" : "livepatch",
         "available" : true
      },
      {
         "available" : true,
         "name" : "support"
      }
   ],
```

hich gives us a traceback. So, we can't land this branch to rely on

Reply:
It's still not clear to me _why_ we want to land this code before it can possibly do anything; if the contract server never implements the thing we've asked for, then we're just introducing dead code. Do we lose anything by deferring landing this specific part of this until the contract server is providing the interface we want to work against?

(The existence of an identically-named key elsewhere means we can _assume_ it will be the same, but there's no guarantee that the implementations won't be different on the backend. For example, your initial attach example doesn't include any `description` elements, so is that _definitely_ the same format as we would get? I don't think we can say that positively without seeing the (not yet completed) implementation in the refresh route in the contract server.)

I think I'd like to see us split this PR up into two pieces: one which relies on the routes we have today (and implements the display logic), and then another which drops the dependency on `get_available_resources` once we have an alternative. Is that reasonable?

Reply:
This code path is exercised on initial attach because initial attach machine-token.json contains availableResources so I think it's reasonable to leave it it. And, we still need the get_available_resources call for the case when we've run "ua enable <anything>" or "ua refresh" where the refreshed machine-token.json doesn't contain available_resources. I'd like the improved performance on ua attach <token> by avoiding this unnecessary v1/resources GET since we don't need it for the default install/attach case.

Reply:
I'm still not fully sold on this. I hadn't realised that the contract server is _already_ inconsistent in its machine token responses when I wrote the above, but I think that further supports my position of "we shouldn't rely on them becoming consistent in future".

Could you split this PR up into the two separate changes it contains so we can continue this discussion without blocking most of this change from landing? (Or at least present some reason why we _can't_ split it up?)